universal_silabs_flasher/cpc.py

```diff
@@ -297,6 +297,8 @@ def data_received(self, data: bytes) -> None:
             except BufferTooShort:
                 break
             except ValueError as e:
+                if self._buffer.find(b"\r\n"):
```

Comment:
What are the contents of the buffer that cause this infinite loop?

Reply:
Ah, no, never mind. This is a bug with the subsequent couple of lines. It should be:

```python
_LOGGER.debug("Failed to parse buffer %r: %r", self._buffer, e)

if bytes([cpc_types.FLAG]) not in self._buffer:
    self._buffer.clear()
    break

self._buffer = self._buffer[
    self._buffer.find(bytes([cpc_types.FLAG])) :
]
```

Reply:
Will be fixed by https://github.com/NabuCasa/universal-silabs-flasher/pull/86.